src/core/backend/pearl/core.rs

```diff
@@ -0,0 +1,614 @@
+        debug!("PUT[alien][{}] to pearl backend", key);
+
+        let alien_dir = self.alien_dir.clone();
+        Put({
+            async move {
+                let result = alien_dir
+                    .write(key, Box::new(data))
+                    .map(|r| {
+                        r.map(|_ok| BackendPutResult {}).map_err(|e| {
+                            debug!(
+                                "PUT[alien][{}], error: {:?}",
+                                key, e
+                            );
+                            e
+                        })
+                    })
+                    .await;
+                if Self::check_write_error(result.as_ref().err())
+                    && alien_dir.try_reinit().await.unwrap() {
+                    //TODO panic if failed lock?
+                    let _ = alien_dir.reinit_storage().await;
+                }
+                result
+            }
+                .boxed()
+        })
+    }
+
+    fn get(&self, disk_name: String, vdisk_id: VDiskId, key: BobKey) -> Get {
+        debug!(
+            "Get[{}][{}][{}] from pearl backend",
+            disk_name, vdisk_id, key
+        );
+
+        let vdisks = self.vdisks.clone();
+        Get({
+            let vdisk = vdisks.iter().find(|vd| vd.equal(&disk_name, &vdisk_id));
+            if let Some(disk) = vdisk {
+                let d_clone = disk.clone(); // TODO remove copy of disk. add Box?
+                async move {
+                    let result = d_clone
+                        .read(key)
+                        .map(|r| {
+                            r.map(|data| BackendGetResult { data }).map_err(|e| {
+                                debug!(
+                                    "GET[{}][{}][{}], error: {:?}",
+                                    disk_name, vdisk_id, key, e
+                                );
+                                e
+                            })
+                        })
+                        .await;
+                    if Self::check_read_error(result.as_ref().err())
+                        && d_clone.try_reinit().await.unwrap() {
+                        //TODO panic if failed lock?
+                        let _ = d_clone.reinit_storage().await;
+                    }
+                    result
+                }
+                    .boxed()
+            }
+            else {
+                debug!(
+                    "GET[{}][{}][{}] to pearl backend. Cannot find storage",
+                    disk_name, vdisk_id, key
+                );
+                err03(backend::Error::VDiskNoFound(vdisk_id)).boxed()
+            }
+        })
+    }
+
+    fn get_alien(&self, _vdisk_id: VDiskId, key: BobKey) -> Get {
+        debug!("Get[alien][{}] from pearl backend", key);
+
+        let alien_dir = self.alien_dir.clone();
+        Get({
+            async move {
+                let result = alien_dir
```

Comment:
can we extract method for operation on vdisk (same code is used for regular vdisk and alien vdisk.